### PR TITLE
RJS-540: Use CMake (and Monorepo)

### PR DESCRIPTION
**Pull request by fealebenpae**
## What, How & Why?

This closes # ???

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 🚦 Tests
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

**Comment by blagoev** [comment]
I guess we should be merging this in `v11`  branch or something

**Comment by fealebenpae** [comment]
Does realm-js normally use something like a develop branch?

File: package.json [hunk 1]
```diff
@@ -39,29 +39,27 @@
...
     "set-version": "scripts/set-version.sh",
     "lint": "eslint",
     "test": "scripts/test.sh",
-    "package": "npm ci --build-from-source=realm && node-pre-gyp package ${npm_config_arch:+--target_arch=${npm_config_arch}}",
-    "install": "node-pre-gyp install --fallback-to-build -j 8 ${npm_config_arch:+--target_arch=${npm_config_arch}}",
-    "rebuild": "npm run isWin -s && SET REALMJS_USE_DEBUG_CORE=1 && npm run rebuild-changes || REALMJS_USE_DEBUG_CORE=1 && npm run rebuild-changes",
-    "build": "npm run isWin -s && SET REALMJS_USE_DEBUG_CORE=1 && npm run build-changes || REALMJS_USE_DEBUG_CORE=1 && npm run build-changes",
-    "build-changes": "node-pre-gyp build --debug -j 8 ${npm_config_arch:+--target_arch=${npm_config_arch}}",
-    "rebuild-changes": "node-pre-gyp rebuild --debug -j 8 ${npm_config_arch:+--target_arch=${npm_config_arch}} && cd tests && npm install",
+    "package": "npm ci --ignore-scripts && prebuild",
```
Comment (blagoev) [inline package.json:52]: shouldn't this be && `npx prebuld`

**Review by blagoev — APPROVED** [review]
great attention to details

File: package.json (hunk 1 above)
Comment (blagoev) [reply package.json:52]: ignore

**Comment by blagoev** [comment]
I guess just create a `develop` branch to merge this PR into

**Comment by bmunkholm** [comment]
@fealebenpae @blagoev What's left to finish this?

**Comment by fealebenpae** [comment]
@bmunkholm there is a known issue where building the Node.js addon for Windows in Release produces a binary that crashes on startup. I'm trying to convince the CI build scripts to build in Debug on Windows just so we can have tests running on that platform still. As soon as the Windows build passes I'm going to merge this so that work on the new data types can begin against the Node.js addon while @blagoev figures out the Windows Release issue, and works to enable the React Native builds as well.

**Comment by fealebenpae** [comment]
We got to a point where the Node.js and Electron tests run and even pass. React Native work will be done separately.

**Pull request merged**